### ISSUE TITLE
FileTarget: Archiving without a pattern creates one archive, then discards all logs

### ISSUE THREAD

**Issue by bhaeussermann** [NLog/NLog#1108]
When one does not include a pattern in the `archiveFileName` property of a `FileTarget` and set `maxArchiveFiles` to a value larger than 1, the target will create one archive file when the log fills up. When the log fills up again afterwards, no archive file is created, and all future logs are discarded.

The code in `FileTarget` seems to handle this case by adding sequential numbers to the end of the archive file name, but the code does not work right.

See example target config:

```
<target name="file" xsi:type="File"
        layout="${longdate} - ${level:uppercase=true}: ${message}"
        fileName="C:\temp\log.txt"
        archiveFileName="C:\temp\archive.txt"
        archiveAboveSize="500"
        archiveNumbering="Rolling"
        maxArchiveFiles="10" />
```

**Comment by 304NotModified** [comment]
Thanks (also for the PR of course!), I was thinking this issue was reported before - but I didn't found it yet

**Comment by 304NotModified** [comment]
Fixed by #1109
